src/Types/BlankFunctionTypeSpecifyingExtension.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\Larastan\Types;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Analyser\SpecifiedTypes;
+use PHPStan\Analyser\TypeSpecifier;
+use PHPStan\Analyser\TypeSpecifierAwareExtension;
+use PHPStan\Analyser\TypeSpecifierContext;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\Constant\ConstantArrayType;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\FunctionTypeSpecifyingExtension;
+use PHPStan\Type\NullType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\UnionType;
+
+final class BlankFunctionTypeSpecifyingExtension implements FunctionTypeSpecifyingExtension, TypeSpecifierAwareExtension
+{
+    /** @var TypeSpecifier */
+    private $typeSpecifier;
+
+    public function isFunctionSupported(
+        FunctionReflection $functionReflection,
+        FuncCall $node,
+        TypeSpecifierContext $context
+    ): bool {
+        return $functionReflection->getName() === 'blank' && ! $context->null();
+    }
+
+    public function specifyTypes(
+        FunctionReflection $functionReflection,
+        FuncCall $node,
+        Scope $scope,
+        TypeSpecifierContext $context
+    ): SpecifiedTypes {
+        $expr = $node->getArgs()[0]->value;
+        $typeBefore = $scope->getType($expr);
+
+        $falseyTypes = $this->getFalseyTypes();
+
+        if ($this->isCalledOnIterable($typeBefore)) {
```

Comment:
I think we can just use `! $typeBefore->isIterable()->no()` inside the if condition here. No need for the `isCalledOnIterable` logic.

Reply:
i am getting: array{}|null for the iterable inside test inside the IF if only do this: 
 if ($typeBefore->isIterable()->yes() && (! $typeBefore instanceof ArrayType))

Reply:
```suggestion
        if (! $typeBefore->isIterable()->no()) {
```

Reply:
yeah tried that, the array and isIterable test is failing after this.